src/core_modules/capture-core/components/Pages/Search/SearchForm/SearchForm.component.js

```diff
@@ -38,21 +38,53 @@ export const getStyles = (theme: Theme) => ({
     },
 });
 
+const useFormDataLifecycle = (
+    searchGroupForSelectedScope,
+    addFormIdToReduxStore,
+    removeFormDataFromReduxStore,
+) =>
+    useEffect(() => {
+        // in order for the Form component to render
+        // a formId under the `forms` reducer needs to be added.
+        searchGroupForSelectedScope
+            .forEach(({ formId }) => {
+                addFormIdToReduxStore(formId);
+            });
+        // we remove the data on unmount to clean the store
+        return () => searchGroupForSelectedScope
+            .forEach(({ formId, searchForm }) => {
+                removeFormDataFromReduxStore(formId);
+
+                Array.from(searchForm.sections.entries())
+                    .map(entry => entry[1])
+                    .forEach(({ id }) => {
+                        removeFormDataFromReduxStore(`${formId}-${id}`);
```

Comment:
I would suggest that we call `removeFormDataFromReduxStore` only once for each form (and deal with the sections in the reducer)

Reply:
Now is being called once. The logic is handled in the container. Perhaps we can avoid logic into the reducer part of the application when it can be handled into the containers.

Reply:
When I'm dispatching the action `removeFormData` I expect all form data for the form to be wiped, that includes all sections of that form. If we want to remove section data only we could create a separate action for that (not needed at the moment). Doesn't that make sense?

Reply:
Yeah I might understand why this is confusing. And I agree it is confusing..

What this iteration does is not removing each one of the fields individually, but basically it constructs some ids (see image). Then it triggers the `removeFormData` to remove these formIds from the store.So in other worlds line 61 will remove the object `searchForm-${id}-#MAIN#` from the store.

I agree this is confusing, but at that time I thought it is necessary. I will find a better way of doing that now. Thanks for raising this

![image](https://user-images.githubusercontent.com/4181674/93176553-1aeb1800-f73a-11ea-878b-9d6021a44902.png)